frontend/src/components/QuestionnairePreview/QuestionnaireList.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import { IQuestion } from 'models/forms/Questions/IQuesion';
+import { ISection } from 'models/forms/Sections/types';
+import QuestionCard from 'components/QuestionnaireOrderDraggableView/QuestionCard';
+import { Header } from 'semantic-ui-react';
+import styles from "./styles.module.sass";
+import { connect } from 'react-redux';
+import { updateQuestionsOrderRoutine } from 'sagas/sections/routines';
+
+interface IIndex  {
+    sectionId: string;
+    questions: IIndexObject[];
+}
+
+interface IIndexObject  {
+    questionId: string;
+    index: number;
+}
+
+interface ISectionQuestionListProps {
+    sectionId: string;
+    questions: IQuestion[];
+    indexQuestions(questions: IIndex): void;
+    handleMoveQuestionToSection(sectionId: string, question: IQuestion, prevSectionId: string): void;
+
+}
+
+const SectionQuestionList: React.FC<ISectionQuestionListProps> = ({
+    sectionId,
+    questions,
+    indexQuestions,
+    handleMoveQuestionToSection
+  }) => {
+    const [questionCards, setQuestionCards] = useState<IQuestion[]>([]);
+  
+    const indexQuestionsHandler = () => {
+      const rst = questions.map((card, i) => { return { questionId: card.id, index: i }; });
+      indexQuestions({sectionId: sectionId,  questions: rst});
+    };
+  
+    useEffect(() => {
+      setQuestionCards(questions);
+    }, [questions]);
+  
+    const moveCard = useCallback(
+      (dragIndex: number, hoverIndex: number) => {
+        const dragCard = questionCards[dragIndex];
+        const updCards = questionCards.slice();
+        updCards.splice(dragIndex, 1);
```

Comment:
please, try to avoid mutable operations if possible

Reply:
It`s not my code, I used existing drag-n-drop logic for changing the position of question inside one chapter.